### PR TITLE
Test controller refactor

### PR DESCRIPTION
**Pull request by fionnbharra**
This is a wip PR, I won't merge it. 

I'm trying to make the test controller a bit easier to read by refactoring into smaller private methods. Would be nice to get a rubyists opinion on where to go next with this. Perhaps @alkesh or @andypearson could comment.

### CODE REVIEW

File: app/controllers/tests_controller.rb [hunk 1]
```diff
@@ -34,90 +34,112 @@ def update
...
-    @test.save!
+  end
 
-    # get the width/height of both the baseline and the test image
-    baseline_screenshot_details = ImageGeometry.new(@test.screenshot_baseline.path)
-    test_screenshot_details = ImageGeometry.new(@test.screenshot.path)
+  def create_canvas(test)
     # create a canvas using the baseline's dimensions
-    canvas = Canvas.new(baseline_screenshot_details, test_screenshot_details)
-    @test.dimensions_changed = canvas.dimensions_differ
```
Comment (fionnbharra) [inline app/controllers/tests_controller.rb]: @nickdunn Do you know if dimensions_changed is used anywhere? I don't think it is. Ok to remove?

Comment (nickdunn) [inline app/controllers/tests_controller.rb]: This just flags whether we had to resize the canvas, but yeah not really necessary any more. Can go.

**Comment by nickdunn** [comment]
:shipit:

**Comment by andypearson** [comment]
The paths, `baseline`, `test` and `diff` seem to always be used together, passed around from place to place. This suggests that you could extract an object that actually does the comparison lifting and clean-up. (`ScreenshotComparison.new(@test)`?)

I've tried to cobble an example together by my brain is falling out at the moment. I'll see if I can do something soon though, but for now I'd be happy to merge this, it's certainly a step in the right direction.

**Comment by fionnbharra** [comment]
Thanks @andypearson. I'll merge this and attempt what you're suggesting in another branch.

**Pull request merged**